tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```
@@ -0,0 +1,189 @@
+            {
+                result = ex.Message;
+            }
+            catch (Exception ex)
+            {
+                exception = ex.Message;
+            }
+
+            //Assert
+            Assert.IsTrue(string.IsNullOrEmpty(exception));
+            Assert.AreSame(result, expectedClientException);
+        }
+
+        [DataTestMethod]
+        [DataRow(AzureResource.webapp, "", DisplayName = "System Identity Web App")]
+        [DataRow(AzureResource.function, "", DisplayName = "System Identity Function App")]
+        [DataRow(AzureResource.webapp, UserAssignedClientID, DisplayName = "User Identity Web App")]
+        [DataRow(AzureResource.function, UserAssignedClientID, DisplayName = "User Identity Function App")]
+        [DataRow(AzureResource.webapp, Mi_res_id, DisplayName = "ResourceID Web App")]
+        [DataRow(AzureResource.function, Mi_res_id, DisplayName = "ResourceID Function App")]
+        public async Task AcquireMSITokenAsync(AzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            AuthenticationResult result = null;
+
+            //Set the Environment Variables
+            bool isEnvironmentVariableSet = await SetEnvironmentVariablesAsync(azureResource)
+                .ConfigureAwait(false);
+
+            if (!isEnvironmentVariableSet)
+            {
+                Assert.Fail("AcquireMSITokenForWebAppAsync failed to set environment variables.");
+            }
+
+            //form the URI 
+            string uri = s_baseURL + $"GetMSIToken?azureresource={azureResource}&uri=";
+
+            //Create CCA with Proxy
+            IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+            //Act
+            try
+            {
+                result = await cca.AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity(userIdentity)
+                    .ExecuteAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
```

Why don't you just let the exception propagate and fail the test?